example/server.rb

```diff
@@ -82,10 +82,25 @@ def loop_over_queries(queries, haystack)
   def search(needle, haystack)
     field, type, value = needle.values_at(*%w(field type value))
     case type.to_sym
-    when :date, :date_relative
+    when :date
       haystack.select do |hay|
         (Date.parse(value["from"])..Date.parse(value["to"])) === hay.send(field)
       end
+    when :date_relative
+      puts value
+      relative_value = value['value']
+      if(!relative_value.nil? && relative_value != '')
+        # TODO: Actually calculate relative dates from selection
```

Comment:
Need to actually implement this to get the correct results, which is slightly more annoying in plain ruby than in rails.